### PR TITLE
chore: Add test for `dsn-file://` catalog urls

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/7715

I broke production 😭  because I broke `dsn-file` handling but didn't realize it. So let's add a test, as suggested by @domodwyer  so I don't do that again

### CODE REVIEW

File: iox_catalog/src/postgres.rs [hunk 1]
```diff
@@ -432,6 +432,18 @@ async fn new_raw_pool(
     Ok(pool)
 }
 
+/// Parse a postgres catalog dsn, handling the special `dsn-file://`
+/// syntax (see [`new_pool`] for more details).
+///
+/// Returns an error if the dsn-file could not be read correctly.
+pub fn parse_dsn(dsn: &str) -> Result<String, sqlx::Error> {
```
Comment (alamb) [inline iox_catalog/src/postgres.rs:439]: refactored this into its own function so I could call it in `test_helpers_end_to_end/src/database.rs`

**Comment by alamb** [comment]
I verified this test fails when I revert the revert c9cfe05f8

```
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Server command failed: Error in ingester2 subcommand: catalog DSN error: Unknown Catalog DSN dsn-file:///var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpnEh5nx/catalog.dsn. Expected a string like 'postgresql://postgres@localhost:5432/postgres' or 'sqlite:///tmp/catalog.sqlite'

```

**Review by wolffcm — APPROVED** [review]
This looks good to me.

**Review by NGA-TRAN — APPROVED** [review]
The test looks good to me

**Pull request merged**